src/libstd/sys/vxworks/fd.rs

```diff
@@ -93,19 +87,13 @@ impl FileDesc {
     }
 
     pub fn write_at(&self, buf: &[u8], offset: u64) -> io::Result<usize> {
```

Comment:
This function is expected to be thread-safe, this implementation of it is not. To implement this function you will need locking somewhere.